File: Homework/themoviedb/src/main/java/com/example/themoviedb/repository/MovieRepositoryImpl.java
package com.example.themoviedb.repository;

import androidx.annotation.NonNull;

import com.example.themoviedb.models.MovieResponse;
import com.example.themoviedb.network.clients.MovieDBClient;

import io.reactivex.Observable;

public class MovieRepositoryImpl implements MovieRepository {

    @NonNull
    private final MovieDBClient client;

    public MovieRepositoryImpl(@NonNull MovieDBClient client) {
        this.client = client;
    }

    @Override
    public Observable<MovieResponse> getMovies(@NonNull String searchString) {
        return client.getMovies(searchString);
    }
}

File: Homework/themoviedb/src/main/java/com/example/themoviedb/repository/MovieRepositoryImpl.kt
package com.example.themoviedb.repository

import com.example.themoviedb.models.MovieResponse
import com.example.themoviedb.network.clients.MovieDBClient

import io.reactivex.Observable

class MovieRepositoryImpl(private val client: MovieDBClient) : MovieRepository {

    override fun getMovies(searchString: String): Observable<MovieResponse> {
        return client.getMovies(searchString)
    }
}